sympy/geometry/ellipse.py

```diff
@@ -151,6 +153,12 @@ def __new__(
         if hradius == 0 or vradius == 0:
             return Segment(Point(center[0] - hradius, center[1] - vradius), Point(center[0] + hradius, center[1] + vradius))
 
+        if (hradius.is_complex is True and hradius.is_real is False) or (vradius.is_complex is True and vradius.is_real is False):
```

Comment:
Use `hradius.is_positive is False or vradius.is_positive is False`
I think that the negative radius may also be disallowed. And positive reals is more stronger condition than anything else here.

Reply:
@sylee957 negative `hradius` are used in other functions of sympy.

Reply:
@sylee957 kindly review this after updating the suggested changes:
```
>>> from sympy import *
>>> test("sympy/geometry/tests/test_ellipse")
```
Returns
``` 
 File "/home/shubham/sympy/sympy/geometry/tests/test_ellipse.py", line 404, in test_reflect
    assert e.area == -e.reflect(Line((1, 0), slope=0)).area
  File "/home/shubham/sympy/sympy/geometry/ellipse.py", line 1185, in reflect
    return self.func(c, -self.hradius, self.vradius)
  File "/home/shubham/sympy/sympy/geometry/ellipse.py", line 160, in __new__
    raise GeometryError("Invalid value encountered when computing hradius / vradius.")
sympy.geometry.exceptions.GeometryError: Invalid value encountered when computing hradius / vradius.
```
This was the reason checks were not kept on `hradius` / `vradius`.

Reply:
Okay I didn't know that negative radius was used in 
However, I think that the check can still be simplified to `hradius.is_real is False and vradius.is_real is False` and the trivial cases like infinity can be checked here too

Reply:
@sylee957 for complex infinity (`zoo`) and `hradius.is_real is False and vradius.is_real is False` is already added.

Reply:
I'd correct to `hradius.is_real is False or vradius.is_real is False`.
What do you mean that the complex infinity cannot be handled in this check?

Reply:
CHecks have been kept on  `(hradius.is_complex is True and hradius.is_real is False) or (vradius.is_complex is True and vradius.is_real is False)`. Reason being some errors were raised.

Reply:
Corrected! my Mistake.